packages/js/product-editor/src/components/variations-table/variation-actions-menus/variation-quick-update-menu-item.tsx

```diff
@@ -0,0 +1,98 @@
+/**
+ * External dependencies
+ */
+import { Slot, Fill, MenuItem } from '@wordpress/components';
+import { createElement, Fragment } from '@wordpress/element';
+import {
+	createOrderedChildren,
+	sortFillsByOrder,
+} from '@woocommerce/components';
+
+/**
+ * Internal dependencies
+ */
+import { MenuItemProps } from './types';
+import { QUICK_UPDATE, SINGLE_UPDATE } from './constants';
+
+const DEFAULT_ORDER = 20;
+
+const getGroupName = ( group?: string, isMultipleSelection?: boolean ) => {
+	const name = 'woocommerce-actions-menu-slot';
+	const nameSuffix = isMultipleSelection
+		? `_${ QUICK_UPDATE }`
+		: `_${ SINGLE_UPDATE }`;
+	return group ? `${ name }_${ group }${ nameSuffix }` : name;
+};
+
+const getMenuItem = ( children: React.ReactNode ) => (
+	<MenuItem>
```

Comment:
I tried using the following:

```js
wp.element.createElement(
                wc.productEditor.VariationQuickUpdateMenuItem,
                {
                    group: 'top-level',
                    supportsMultipleSelection: true,
                    onClick: () => alert( 'Plugin top level multiple selection!' ),
                },
                'Plugin top level multiple selection!',
),
```

And my `onClick` didn't work.

I assume it is because we don't actually pass down any props to the `MenuItem`.

Reply:
I'd say to keep things locked up a bit and only support that props we really need right now, like `onClick`. That way, we keep the flexibility to change the underlying implementation in the future. And, we prevent extensions from using props that would result in poor UX.

Reply:
Added in [5903136](https://github.com/woocommerce/woocommerce/pull/43441/commits/59031365dfc7104f2b219ed12e8e93f1eacd4e3f)